### PR TITLE
Kerrat lecture 7. Students and marks + Pizza + Book + author

### PR DESCRIPTION
**Pull request by KERRAT**

### CODE REVIEW

File: lecture-7/Book_author/Author.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (nromanen) [inline lecture-7/Book_author/Author.rb:1]: add eql? method and used it in exists?(name, surname)

File: lecture-7/Book_author/Author.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def exists?(name, surname)
+    ex = false
+    @@authors.each { |el|
+      if el.name == name && el.surname == surname
+        ex = true
+        break
+      end
+    }
```
Comment (nromanen) [inline lecture-7/Book_author/Author.rb]: it's better to use any? and avoid using flag ex

File: lecture-7/Book_author/Author.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
+class Author
+  attr_reader :name, :surname, :patronymic
+  @@authors = []
+  @@pattern_name = /^\p{Lu}\p{Ll}{1,14}(-(\p{Lu}\p{Ll}{1,14}$)|$)/
+  @@pattern = /^\p{Lu}\p{Ll}{1,14}$/
+
+  def initialize(name, surname, patronymic ="")
+    unless name.match(@@pattern_name) == nil || surname.match(@@pattern) == nil
```
Comment (nromanen) [inline lecture-7/Book_author/Author.rb]: use if statement in this case

File: lecture-7/Book_author/Book.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+
+class Book
+  attr_reader :name, :authors, :issued, :year
+  @@pattern_name = /[\p{L}\d\-.,;]{1,10}/
+  @@pattern_year = /\d{4}/
+
+  def initialize(name, authors, issued, year)
```
Comment (nromanen) [inline lecture-7/Book_author/Book.rb:7]: refactor this method without authors_pass flag

File: lecture-7/Book_author/Book.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      @year = year
+      @issued = issued
+    end
+
+
+
+
+  end
+
+  def exists?(name, surname)
```
Comment (nromanen) [inline lecture-7/Book_author/Book.rb]: replace each method with any? method

File: lecture-7/Student/Student.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  public
+
+  def initialize(name, marks = [])
+    @name = name
+    @marks = marks
+  end
+
+  def exists?(subject_title)
+    ex = false
+    @marks.each { |el|
```
Comment (nromanen) [inline lecture-7/Student/Student.rb:17]: use any? method instead of each

File: lecture-7/Book_author/Author.rb (hunk 1 above)
Comment (KERRAT) [reply lecture-7/Book_author/Author.rb:1]: def eql?(name, surname)
    self.name == name && self.surname == surname
  end

File: lecture-7/Book_author/Book.rb (hunk 4 above)
Comment (KERRAT) [reply lecture-7/Book_author/Book.rb:7]: done using  @authors = nil

**Pull request merged**